### ISSUE TITLE
Allow rate limiting

### ISSUE THREAD

**Issue by Marthisdil** [alexta69/metube#130]
YT-DLP doesn't support it directly, but the ratelimit ability is used in the downloader (https://github.com/yt-dlp/yt-dlp/blob/master/yt_dlp/YoutubeDL.py#L176 (line 466))

Even better would be a little scheduler where you can rate limit during, say, working hours, but late night, have it be at a different limit

**Comment by alexta69** [comment]
You can set the option via the YTDL_OPTIONS environment variable, see the README. About the scheduler, I'm not sure how common the use case it. Leaving it here for future feedback.

**Comment by tekwarfare** [comment]
Not directly relevant, but I'm leaving how to limit the download speed here. I believe it would be useful to add this as an example in the _Configuration via environment variables_ section as I struggled for a while with all the permutations.
```
Config Type: Variable
Name: YTDL_OPTIONS
Key: YTDL_OPTIONS
Value: {"ratelimit": 3000000}
```
Speed is in bytes/s so 3000000B/s = 3.00MB/s

**Comment by ericnitschke** [comment]
@alexta69 this was marked as completed but I don't see the option in the web UI. Is changing YTDL_OPTIONS  the only method? thanks in advance.

**Comment by alexta69** [comment]
@ericnitschke yes, there isn't another way, as with most things configurable via YTDL_OPTIONS.. the possibilities there number in the hundreds, so it's unlikely they'll be added to the UI. We're adding to the UI mostly stuff that changes the behavior of MeTube, rather than the behavior of yt-dlp within it (which we leave to configure via YTDL_OPTIONS).

**Comment by jtmusson** [comment]
> ```
> Config Type: Variable
> Name: YTDL_OPTIONS
> Key: YTDL_OPTIONS
> Value: {"ratelimit": 3000000}
> ```
> 
> Speed is in bytes/s so 3000000B/s = 3.00MB/s

Coming here to say thank you for noting this, exactly what I was looking for to rein in ytdl a bit and not saturate the internet connection during downloads. It's worth also being aware for anyone trying this that the speed entered here is in MB/s whereas your connection is usually rated in Mbps. The conversion factor is approximately *8, so a ratelimit of 3.00 MB/s would translate to 24 Mbps (roughly).
